### PR TITLE
search feature

### PR DESCRIPTION
**Pull request by darkskygit**
this pr add basic indexing quality local search feature into @blocksuite/store

the indexing service will refresh index all prop:text and prop:title in a block when they add/update/remove, at this time indexing feature support ascii words and cjk tokenize, if user use non-chromium core, tokenize function will fallback to char based tokenize

image

in future we will implement a native version in rust with same apis, and must allow add more custom props from outside

### CODE REVIEW

File: packages/store/src/search.ts [hunk 1]
```diff
@@ -0,0 +1,146 @@
+import { Document as DocumentIndexer, DocumentSearchOptions } from 'flexsearch';
+import { Doc, Map as YMap, Text as YText } from 'yjs';
+import type { YBlock } from './space';
+
+export type QueryContent = string | Partial<DocumentSearchOptions<boolean>>;
+
+function tokenizeZh(text: string) {
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  const tokenizer = Intl?.v8BreakIterator;
```
Comment (doodlewind) [inline packages/store/src/search.ts:10]: Can we replace this non-standard API with standardized [Intl.Segmenter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Segmenter)?

File: packages/store/src/store.ts [hunk 2]
```diff
@@ -51,6 +59,8 @@ export class Store {
       spaceId,
       new Space(spaceId, this.doc, this.awareness, this.idGenerator)
     );
+    this._indexer.onCreateSpace(spaceId);
```
Comment (doodlewind) [inline packages/store/src/store.ts:62]: What should we do after the space is updated afterward?

File: packages/store/src/search.ts (hunk 1 above)
Comment (darkskygit) [reply packages/store/src/search.ts:10]: v8BreakIterator can support chromium after 2018, Intl.Segmenter add official support after 2020/11, maybe we can make v8BreakIterator as a fallback

File: packages/store/src/store.ts (hunk 2 above)
Comment (darkskygit) [reply packages/store/src/store.ts:62]: seem we cannot observe root map change (doc update event / abstract type event)

File: packages/store/src/search.ts (hunk 1 above)
Comment (doodlewind) [reply packages/store/src/search.ts:10]: Browser support after 2020 looks promising. I think it'll be fine to support browser versions within a year for the oversea market, which is what [photopea](https://www.photopea.com/) is doing.

**Pull request merged**